src/global.cpp

```diff
@@ -36,27 +36,99 @@ int symPACK_Rank(int * rank){
 }
 
 
+#ifdef CUDA_MODE
+extern "C"
+void compute_device_alloc_size() {
+  int n_gpus = upcxx::gpu_default_device::device_n();
+  if (n_gpus==0) {
+    std::cerr<<"Error: found no CUDA devices"<<std::endl;
+    abort();
+  }
+
+  int tasks_per_node = upcxx::local_team().rank_n();
+ 
+  /* NOTE: This algorithm is known to underestimate the amount of space 
+   * available on nodes that have more than one physical GPU, so using the -gem_mem argument is encouraged
+   * in such cases. See README.md for details.
+   */ 
+  size_t alloc_size, free, total;
+  CUDA_ERROR_CHECK(cudaMemGetInfo(&free, &total));
+  alloc_size = (size_t)(free*0.8) / (std::max(tasks_per_node, n_gpus) / n_gpus);
+  symPACK::gpu_alloc_size = alloc_size;
+}
+ 
+extern "C"
+void symPACK_cuda_setup(symPACK::symPACKOptions optionsFact) {
+  if (optionsFact.gpu_alloc_size != 0) {
+    symPACK::gpu_alloc_size = optionsFact.gpu_alloc_size;  
+  }
+
+  symPACK::fallback_type = optionsFact.fallback_type; 
+
+  symPACK::gpu_allocator = upcxx::make_gpu_allocator<upcxx::gpu_default_device>(symPACK::gpu_alloc_size);
+
+  int gpu_id = symPACK::gpu_allocator.device_id();
+  cudaSetDevice(gpu_id);
+  
+  cublasHandle_t handle;
+  symPACK::cublas_handler= handle; 
+  CUBLAS_ERROR_CHECK(cublasCreate(&symPACK::cublas_handler));
+
+  cusolverDnHandle_t cusolver_handle;
+  symPACK::cusolver_handler = cusolver_handle;
+  CUSOLVER_ERROR_CHECK(cusolverDnCreate(&symPACK::cusolver_handler));
+  
+  symPACK::gpu_block_limit = optionsFact.gpu_block_limit;
+  symPACK::trsm_limit = optionsFact.trsm_limit;
+  symPACK::potrf_limit = optionsFact.potrf_limit;
+  symPACK::gemm_limit = optionsFact.gemm_limit;
+  symPACK::syrk_limit = optionsFact.syrk_limit;
+  symPACK::gpu_solve = optionsFact.gpu_solve;
+  
+  if (optionsFact.gpu_verbose) {
+    if (upcxx::rank_me()==0) {
+        std::cout<<"========= GPU CONFIGURATION OPTIONS ========="<<std::endl;
+        std::cout<<"-gpu_mem: "<<symPACK::gpu_alloc_size<<" bytes"<<std::endl;
```

Comment:
Weak request to report in MiB for readability:
```suggestion
        std::cout<<"-gpu_mem: "<<(symPACK::gpu_alloc_size/(double)(1<<20))<<" MiB"<<std::endl;
```
would be even cooler if the `-gpu_mem` command line arg accepted an "M*" or "G*" suffix

Reply:
optional suffix added in [119993d](https://github.com/symPACK/symPACK/pull/12/commits/119993d975ffe5ae5a166d3fad5f72f84f1de936)